### PR TITLE
Add sanitiser for nested request bodies

### PR DESCRIPTION
**Pull request by davidrapson**
Add sanitiser for nested request bodies. Accounts for body-parse urlencoded extended values where data is expanded into nested objects and arrays.

Attempts to only sanitises if the value is string-like to avoid unwanted stringify-ing of other values.

### CODE REVIEW

**Comment by davidrapson** [comment]
[PT 9.8: Strip malicious input (rather than sanitising it) (LOW)](https://trello.com/c/hgfuQZSp/682-pt-98-strip-malicious-input-rather-than-sanitising-it-low)

File: controllers/apply/form-router-next/steps.js [hunk 1]
```diff
@@ -109,9 +110,13 @@ module.exports = function(formId, formBuilder) {
                 currentApplicationData
             } = res.locals;
 
+            const sanitisedBody = sanitiseRequestBody(req.body);
+
+            console.log(sanitisedBody);
```
Comment (mattandrews) [inline controllers/apply/form-router-next/steps.js]: ❌

Comment (mattandrews) [reply controllers/apply/form-router-next/steps.js]: (though wonder if we should log the number of sanitisations made?)

Comment (davidrapson) [reply controllers/apply/form-router-next/steps.js]: 🙈 And good call, added debug level logging to the sanitise method:

image

**Pull request merged**